include/up-cpp/utils/Expected.h

```diff
@@ -22,87 +23,161 @@ namespace uprotocol::utils {
+
+	constexpr const E&& error() const&& noexcept { return std::move(storage); }
+
+	constexpr E&& error() && noexcept { return std::move(storage); }
+};
 
 /// @brief A stripped-down version of std::expected from C++23.
-template <typename T, typename E>
+template <typename E, typename U>
 struct Expected {
-	using value_type = T;
-	using error_type = E;
-
-	constexpr Expected();
-	constexpr Expected(const Expected& other);
-	constexpr Expected(Expected&& other) noexcept;
-
-	constexpr explicit Expected(T&& v);
-	constexpr explicit Expected(E&& e);
-
-	Expected& operator=(const Expected& other);
-	Expected& operator=(Expected&& other);
-
-	Expected& operator=(const T& other);
-	Expected& operator=(T&& other);
-
-	Expected& operator=(const E& other);
-	Expected& operator=(E&& other);
-
-	constexpr explicit operator bool() const noexcept;
-	constexpr bool has_value() const noexcept;
-
-	constexpr T& value() &;
-	constexpr const T& value() const&;
-	constexpr T&& value() &&;
-	constexpr const T&& value() const&&;
-
-	constexpr const E& error() const& noexcept;
-	constexpr E& error() & noexcept;
-	constexpr const E&& error() const&& noexcept;
-	constexpr E&& error() && noexcept;
-
-	constexpr T value_or(T&& default_value) const&;
-	constexpr T value_or(T&& default_value) &&;
-
-	template <typename F>
-	constexpr auto and_then(F&& f) &;
-	template <typename F>
-	constexpr auto and_then(F&& f) const&;
-	template <typename F>
-	constexpr auto and_then(F&& f) &&;
-	template <typename F>
-	constexpr auto and_then(F&& f) const&&;
-
-	template <typename F>
-	constexpr auto or_else(F&& f) &;
-	template <typename F>
-	constexpr auto or_else(F&& f) const&;
-	template <typename F>
-	constexpr auto or_else(F&& f) &&;
-	template <typename F>
-	constexpr auto or_else(F&& f) const&&;
+	std::variant<E, Unexpected<U>> storage;
+
+public:
+	constexpr Expected() { storage = E(); }
+	constexpr Expected(E&& rhs) : storage(std::move(rhs)) {}
+	constexpr Expected(Unexpected<U>&& rhs) : storage(std::move(rhs)) {}
+
+	constexpr Expected(const Unexpected<U>& rhs) {
+		std::get<Unexpected<U>>(storage).storage = std::move(rhs);
+	}
+
+	template <class X = E>
+	constexpr explicit Expected(X&& rhs) {
+		std::get<E>(storage) = std::forward<X>(rhs);
+	}
+
+	template <class X = U>
+	constexpr explicit Expected(Unexpected<X>&& rhs) {
+		std::get<E>(storage) = rhs.error();
+	}
+
+	constexpr Expected(const Expected& rhs) { storage = rhs.storage; }
+
+	constexpr bool has_value() const noexcept {
+		return std::holds_alternative<E>(storage);
+	}
+
+	constexpr explicit operator bool() const noexcept {
+		return std::holds_alternative<E>(storage);
+	}
+
+	template <class X>
+	constexpr E value_or(X&& v) const& noexcept {
+		return has_value() ? std::get<E>(storage) : v;
+	}
+
+	constexpr const E& value() const& {
+		if (!has_value())
+			throw BadExpectedAccess(
+			    "Attempt to access value() when unexpected.");
+		return std::move(std::get<E>(storage));
```

Comment:
This shouldn't move since it's the const version
```suggestion
		return std::get<E>(storage);
```

Reply:
This will be fixed in the next push.